spark-java/src/main/java/com/baeldung/sparkjava/UserService.java

```diff
@@ -0,0 +1,13 @@
+package com.baeldung.sparkjava;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+public interface UserService {
+    public void addUser (User user) ;
+    public Collection<User> getUsers () ;
+    public User getUser (String id) ;
+    public User editUser (String id, HashMap userArg) throws Exception;
```

Comment:
Changes to Map and Created a new Exception Class. Changed the flow of the function a bit. 

About passing the JSON itself, Actually it passes the JSON only, If you see line 34 of SparkRestExample.java (put("/users/:id", (request, response) -> {..) it converts JSON to map and hence map only has fields which are sent to be edited.

I was looking for better ways to do this since inception and have also shared my concerned earlier for the same. 

Let me know if we can still make this better after this above edits i've done in the editUser function.